internal/insights/store/store.go

```diff
@@ -852,6 +853,31 @@ func (s *Store) LoadAggregatedIncompleteDatapoints(ctx context.Context, seriesID
 	})
 }
 
+// This is a variant of LoadAggregatedIncompleteDatapoints, but it retains the repoId so that
+// the graphql resolvers can resolve the repository.l
+func (s *Store) LoadIncompleteDatapoints(ctx context.Context, seriesID int) (results []IncompleteDatapoint, err error) {
+	if seriesID == 0 {
+		return nil, errors.New("invalid seriesID")
+	}
+
+	q := "select reason, time, repo_id from insight_series_incomplete_points where series_id = %s group by reason, time, repo_id;"
```

Comment:
If returning a set of repositories that failed, we could instead do something like:

```suggestion
	q := "select reason, time, ARRAY_AGG(repo_id) from insight_series_incomplete_points where series_id = %s group by reason, time;"
```